code/modules/modular_armor/modular.dm

```diff
@@ -41,59 +41,154 @@
 	gas_transfer_coefficient = 1
 
 	actions_types = list(/datum/action/item_action/toggle)
+	///Assoc list of available slots.
+	var/list/attachments_by_slot = list(
+		ATTACHMENT_SLOT_CHESTPLATE,
+		ATTACHMENT_SLOT_SHOULDER,
+		ATTACHMENT_SLOT_KNEE,
+		ATTACHMENT_SLOT_MODULE,
+		ATTACHMENT_SLOT_STORAGE,
+	)
+	///Typepath list of allowed attachment types.
+	var/list/attachments_allowed = list(
```

Comment:
both of those could be cached IMO, but it's non blocking if you don't want to do it